core/src/test/scala/unit/kafka/admin/DescribeConsumerGroupTest.scala

```diff
@@ -149,7 +154,130 @@ class DescribeConsumerGroupTest extends KafkaServerTestHarness {
+        assignments.get.filter(_.group == group).head.consumerId.exists(!_.trim.equals(ConsumerGroupCommand.MISSING_COLUMN_VALUE)) &&
+        assignments.get.filter(_.group == group).head.clientId.exists(!_.trim.equals(ConsumerGroupCommand.MISSING_COLUMN_VALUE)) &&
+        assignments.get.filter(_.group == group).head.host.exists(!_.trim.equals(ConsumerGroupCommand.MISSING_COLUMN_VALUE))
+      }, "Expected a 'Stable' group status, rows and valid values for consumer id / client id / host columns in describe group results.")
+
+    // cleanup
+    consumerGroupCommand.close()
+    consumerMock.cleanup()
+  }
+
+  @Test
+  def testDescribeExistingGroupWithNoMembersWithNewConsumer() {
+    // mocks
+    props.setProperty("bootstrap.servers", brokerList)
+    props.setProperty("key.deserializer", classOf[ByteArrayDeserializer].getName)
+    props.setProperty("value.deserializer", classOf[ByteArrayDeserializer].getName)
+    val consumerMock = EasyMock.createMockBuilder(classOf[NewShinyConsumer]).withConstructor(Some(topic), None, None, None, props, Long.MaxValue.asInstanceOf[Object]).createMock()
+
+    // stubs
+    val opts = new ConsumerGroupCommandOptions(Array("--bootstrap-server", brokerList, "--describe", "--group", group))
+    val consumerGroupCommand = new KafkaConsumerGroupService(opts)
+
+    // simulation
+    EasyMock.replay(consumerMock)
+
+    // action/test
+    consumerGroupCommand.describeGroup() // there should be a member here
+    consumerMock.cleanup()
+    TestUtils.waitUntilTrue(() => {
+        val (state, assignments) = consumerGroupCommand.describeGroup()
+        state == Some("Empty") &&
+        assignments.isDefined &&
+        assignments.get.filter(_.group == group).size == 1 &&
+        assignments.get.filter(_.group == group).head.consumerId.exists(_.trim.equals(ConsumerGroupCommand.MISSING_COLUMN_VALUE)) && // the member should be gone
+        assignments.get.filter(_.group == group).head.clientId.exists(_.trim.equals(ConsumerGroupCommand.MISSING_COLUMN_VALUE)) &&
+        assignments.get.filter(_.group == group).head.host.exists(_.trim.equals(ConsumerGroupCommand.MISSING_COLUMN_VALUE))
+      }, "Expected no active member in describe group results.")
+
+    // cleanup
+    consumerGroupCommand.close()
+  }
+
+  @Test
+  def testDescribeConsumersWithNoAssignedPartitionsWithNewConsumer() {
+    // mocks
+    props.setProperty("bootstrap.servers", brokerList)
+    props.setProperty("key.deserializer", classOf[ByteArrayDeserializer].getName)
+    props.setProperty("value.deserializer", classOf[ByteArrayDeserializer].getName)
+    val consumer1Mock = EasyMock.createMockBuilder(classOf[NewShinyConsumer]).withConstructor(Some(topic), None, None, None, props, Long.MaxValue.asInstanceOf[Object]).createMock()
+    val consumer2Mock = EasyMock.createMockBuilder(classOf[NewShinyConsumer]).withConstructor(Some(topic), None, None, None, props, Long.MaxValue.asInstanceOf[Object]).createMock()
```

Comment:
@hachikuji I was hoping you could take a look at an issue I'm running into if and when you get a chance.
While KIP-88 is open for discussion I spent some time creating some unit tests for this PR.
This particular unit test simply mocks two consumers that consume from the same 1-partition topic and belong to the same consumer group. A similar unit test on old consumers exists earlier in the same file and runs fine. There are also other unit tests above using the new consumer that run fine but they mock only one consumer.

The problem I'm running into is this line that mocks the second consumer and takes a long time to run (I believe for the consumer to join the group that eventually times out) and the consumer group gets corrupted somehow. When I debug and check the status of the group, sometimes it is `Empty`, or `Dead`, or even `Stable` with only one of the consumers. Where it gets stuck in a loop I think is [here](https://github.com/vahidhashemian/kafka/blob/5b1c5fc78477dc1c5cf9e48429d93fde15c0f4e6/clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java#L180) (after [trying to join the group](https://github.com/vahidhashemian/kafka/blob/5b1c5fc78477dc1c5cf9e48429d93fde15c0f4e6/clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java#L307)).

I'm not sure if I'm doing something wrong with the unit test or if I'm hitting some bug. I thought you might know by just looking at it. Thanks.